### PR TITLE
[ADR] Use NextJS for the USSF portal client application

### LINKED ISSUE #24: Write ADR on chosen language(s), framework(s), and architecture for initial portal work.

### PR DESCRIPTION
**Pull request by suzubara**
## Description 

This adds an ADR to use NextJS for the new portal client application.

Fixes #24

### CODE REVIEW

File: docs/adr/0002-nextjs.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Use NextJS for the USSF portal client application
+
+* Status: Proposed
+* Deciders: @suzubara @abbyoung @esacteksab @noahfirth
+* Date: 2021-06-24
+
+*To avoid bikeshedding, this ADR uses the [lightweight ADR template](https://adr.github.io/#lightweight-architectural-decision-records-should-be-adopted) and exists primarily for recording the decision, rather than weighing pros & cons of all considered options.*
```
Comment (esacteksab) [inline docs/adr/0002-nextjs.md:7]: I appreciate this, I _really_ do, but I feel like there is at least value in calling out the other frameworks we _did_ look at, if only briefly. I don't think we need a laundry list of why we didn't choose them, just an acknowledgement that NextJS wasn't our only consideration. Remember, these aren't necessarily for us, but those who come after us to gain awareness of why things are they way they are. If you read this 1, 2, 3 years later as a new person coming onto the project trying to understand "why NextJS?" I have to imagine, confusion may be one of the things you might be feeling.

Comment (suzubara) [reply docs/adr/0002-nextjs.md:7]: 💯 that's totally fair. Now that I'm thinking about it, I think this decision goes hand in hand with the decision to use a headless CMS, so I'll update this ADR with some more of that context as well as the alternatives we considered.

**Review by esacteksab — APPROVED** [review]

**Review by abbyoung — APPROVED** [review]
excellent synthesis! i am excited all over again to work with nextjs 😄

**Review by noahfirth — APPROVED** [review]
This is a really great overview of the decision to use NextJS!

**Pull request merged**